src/backend/marsha/core/tests/test_api_liveregistration.py

```diff
@@ -0,0 +1,2817 @@
+            lti_user_id="56255f3807599c377bf0e5bf0723DIFF",
+            consumer_site=video.playlist.consumer_site,
+        )
+
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["context_id"] = str(video.playlist.lti_id)
+        # token with a different lti_user_id
+        jwt_token.payload["user"] = {
+            "email": None,
+            "id": "56255f3807599c377bf0e5bf072359fd",
+        }
+
+        response = self.client.get(
+            f"/api/liveregistrations/{liveregistration.id}/",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+
+        self.assertEqual(response.status_code, 404)
+
+    def test_api_liveregistration_read_token_user_with_email_none_context_with_nok_r_consumer_none(
+        self,
+    ):
+        """Token with no email can't read its registration if consumer site is different.
+
+        LTI users don't necessary have email. Token users with no email in token can only read
+        registration if the registration has the same duo key lti_user_id and consumer_site.
+        lti_user_id are only unique by consumer_site.
+        """
+        video = VideoFactory()
+        # registration with no consumer_site
+        liveregistration = LiveRegistrationFactory(
+            video=video,
+            lti_user_id="56255f3807599c377bf0e5bf072359fd",
+        )
+        other_playlist = PlaylistFactory()
+        # token with different context_id and no email
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["context_id"] = str(other_playlist.lti_id)
+        jwt_token.payload["user"] = {
+            "email": None,
+            "id": "56255f3807599c377bf0e5bf072359fd",
+        }
+
+        response = self.client.get(
+            f"/api/liveregistrations/{liveregistration.id}/",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+
+        self.assertEqual(response.status_code, 404)
+
+    def test_api_liveregistration_read_token_user_with_email_none_context_with_nok_r_consumer(
+        self,
+    ):
+        """Token with no email can't read its registration with different consumer_site.
+
+        LTI users don't necessary have email. Token users with no email in token can only read
+        registration if the registration has the same duo key lti_user_id and consumer_site.
+        lti_user_id are only unique by consumer_site.
+        """
+        video = VideoFactory()
+        # registration with consumer_site
+        liveregistration = LiveRegistrationFactory(
+            video=video,
+            lti_user_id="56255f3807599c377bf0e5bf072359fd",
+            consumer_site=video.playlist.consumer_site,
+        )
+        other_playlist = PlaylistFactory()
+
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        # context_id from another playlist meaning another consumer site
```

Comment:
yes, so it's different than the one we compare it to, consumer_site=video.playlist.consumer_site,
I'll take away the comment if it is confusing.

Reply:
I've left the other sentence that I've used everywhere, you'll tell me if this one is less confusing to you. I can get rid of leading to another consumer site but I think it can help the understanding of the test. You'll tell me

# context_id from another playlist leading to another consumer site